misc-utils/lsclocks.c

```diff
@@ -0,0 +1,366 @@
+		{ "raw",        no_argument,       NULL, 'r' },
+		{ "time",       required_argument, NULL, 't' },
+		{ 0 },
+	};
+
+	setlocale(LC_ALL, "");
+	bindtextdomain(PACKAGE, LOCALEDIR);
+	textdomain(PACKAGE);
+	close_stdout_atexit();
+
+	while ((c = getopt_long(argc, argv, "no:JrVh", longopts, NULL)) != -1) {
+		switch (c) {
+		case 'n':
+			noheadings = true;
+			break;
+		case 'o':
+			outarg = optarg;
+			break;
+		case 'J':
+			json = true;
+			break;
+		case 'r':
+			raw = true;
+			break;
+		case 't':
+			clock = parse_clock(optarg);
+			break;
+		case 'V':
+			print_version(EXIT_SUCCESS);
+		case 'h':
+			usage();
+		default:
+			errtryhelp(EXIT_FAILURE);
+		}
+	}
+
+	if (argv[optind])
+		errtryhelp(EXIT_FAILURE);
+
+	if (clock != -1) {
+		rc = clock_gettime(clock, &now);
+		if (rc)
+			err(EXIT_FAILURE, _("failed to get time"));
+		printf("%ju.%09"PRId32"\n", (uintmax_t) now.tv_sec, (uint32_t) now.tv_nsec);
+		return EXIT_SUCCESS;
+	}
+
+	if (outarg && string_add_to_idarray(outarg, columns, ARRAY_SIZE(columns),
+					    &ncolumns, column_name_to_id) < 0)
+		return EXIT_FAILURE;
+
+	if (!ncolumns) {
+		columns[ncolumns++] = COL_ID;
+		columns[ncolumns++] = COL_NAME;
+		columns[ncolumns++] = COL_HUMAN_NAME;
+		columns[ncolumns++] = COL_TIME;
+		columns[ncolumns++] = COL_ISO_TIME;
+		columns[ncolumns++] = COL_RESOLUTION;
+	}
+
+	scols_init_debug(0);
+
+	tb = scols_new_table();
+	if (!tb)
+		errx(EXIT_FAILURE, _("failed to allocate output table"));
+	scols_table_set_name(tb, "clocks");
+
+	for (i = 0; i < ncolumns; i++) {
+		colinfo = &infos[columns[i]];
+
+		col = scols_table_new_column(tb, colinfo->name, colinfo->whint, colinfo->flags);
+		if (!col)
+			errx(EXIT_FAILURE, _("failed to allocate output column"));
+
+		scols_column_set_json_type(col, colinfo->json_type);
+	}
+
+	for (i = 0; i < ARRAY_SIZE(clocks); i++) {
+		clockinfo = &clocks[i];
+
+		ln = scols_table_new_line(tb, NULL);
+		if (!ln)
+			errx(EXIT_FAILURE, _("failed to allocate output line"));
+
+		/* outside the loop to guarantee consistency between COL_TIME and COL_ISO_TIME */
+		rc = clock_gettime(clockinfo->id, &now);
+		if (rc)
+			now.tv_nsec = -1;
+
+		for (j = 0; j < ncolumns; j++) {
+			switch (columns[j]) {
+				case COL_ID:
+					scols_line_asprintf(ln, j, "%ju", (uintmax_t) clockinfo->id);
+					break;
+				case COL_NAME:
+					scols_line_set_data(ln, j, clockinfo->id_name);
+					break;
+				case COL_HUMAN_NAME:
+					scols_line_set_data(ln, j, clockinfo->name);
+					break;
+				case COL_TIME:
+					if (now.tv_nsec == -1)
+						break;
+
+					scols_line_format_timespec(ln, j, &now);
+					break;
+				case COL_ISO_TIME:
+					if (now.tv_nsec == -1)
+						break;
+
+					rc = strtimespec_iso(&now,
+							ISO_GMTIME | ISO_DATE | ISO_TIME | ISO_T | ISO_DOTNSEC | ISO_TIMEZONE,
+							buf, sizeof(buf));
+					if (rc)
+						errx(EXIT_FAILURE, _("failed to format iso time"));
+					scols_line_set_data(ln, j, buf);
+					break;
```

Comment:
The example in `man clock_gettime` also prints relative time, for example, "8356 days +  7h 30m 59s", what about support also this?

Reply:
It would be nice, for example, for things like `boottime`.